File: assets/samples/whois/password/php/WhoisApiQuery.php
<?php

$domain = 'example.com';
$password = 'your whois api password';
$username = 'your whois api username';

$url = 'http://www.whoisxmlapi.com/whoisserver/WhoisService?'
     . "domainName={$domain}&username={$username}&password={$password}";

print(file_get_contents($url));

File: assets/samples/whois/password/ruby/WhoisApiQuery.rb
require 'open-uri'

domain = 'example.com'
password = 'your whois api password'
username = 'your whois api username'

url = 'http://www.whoisxmlapi.com/whoisserver/WhoisService?'\
    + 'domainName=' + domain + '&username=' +username + '&password=' +password

puts open(url).read